src/main/java/com/aws/greengrass/clientdevices/auth/configuration/SecurityConfiguration.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.configuration;
+
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.util.Coerce;
+import lombok.Getter;
+
+import java.util.Objects;
+
+/**
+ * Represents the security configuration. Acts as an adapter from the GG Topics to the domain.
+ * <p>
+ * |---- configuration
+ * |    |---- security:
+ * |          |---- clientDeviceTrustDurationMinutes: "..."
+ * </p>
+ */
+@Getter
+public final class SecurityConfiguration {
+    private static final Logger logger = LogManager.getLogger(SecurityConfiguration.class);
+    public static final String SECURITY_TOPIC = "security";
+    public static final String CLIENT_DEVICE_TRUST_DURATION_MINUTES_TOPIC = "clientDeviceTrustDurationMinutes";
+    public static final int DEFAULT_CLIENT_DEVICE_TRUST_DURATION_MINUTES = 24;
+    public static final int MIN_CLIENT_DEVICE_TRUST_DURATION_MINUTES = 0;
+    public static final int MAX_CLIENT_DEVICE_TRUST_DURATION_MINUTES = Integer.MAX_VALUE;
```

Comment:
Can be removed. You're reading the value as an int. By definition, there no value larger than `Integer.MAX_VALUE` so clamping will never do anything. I'm guessing the compiler will probably remove this.

Reply:
agreed, removing this max-duration and also the clamping logic. The overflown integer values are now set to minimum trust duration value, zero, with an error log.